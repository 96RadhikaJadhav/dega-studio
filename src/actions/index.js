import { GET_ORGANISATIONS, GET_ORGANISATIONS_SUCCESS } from './types';
import axios from 'axios';

export const getOrganisations = () => {
  return (dispatch, getState) => {};
};

const getOrganisationsSuccess = (todo) => ({
  type: GET_ORGANISATIONS_SUCCESS,
  payload: {},
});
