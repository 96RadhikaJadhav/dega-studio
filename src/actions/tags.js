import axios from '../utils/axios';
import {
  GET_TAGS_SUCCESS,
  GET_TAGS_FAILURE,
  ADD_TAG_FAILURE,
  ADD_TAG_SUCCESS,
  API_ADD_TAG,
  API_GET_TAGS,
  UPDATE_TAG_FAILURE,
  UPDATE_TAG_SUCCESS,
  DELETE_TAG_SUCCESS,
  DELETE_TAG_FAILURE,
  LOADING_TAGS,
} from '../constants/tags';

export const getTags = (query) => {
  return async (dispatch, getState) => {
    let found = false;
    const {
      tags: { req },
    } = getState();

    // map data based on query
    req.forEach((each) => {
      const { limit, page } = each.query;
      if (page === query.page && limit === query.limit) {
        found = true;
        return;
      }
    });

    if (!found) {
      dispatch(loadingSpaces());
      const response = await axios({
        url: API_GET_TAGS,
        method: 'get',
        params: query,
      }).catch((error) => {
        dispatch(getTagsFailure(error.message));
      });
      if (response) {
        dispatch(getTagsSuccess(response.data, query));
      }
    }
  };
};

export const addTag = (data) => {
  return async (dispatch, getState) => {
    dispatch(loadingSpaces());
    const response = await axios({
      url: API_ADD_TAG,
      method: 'post',
      data: { ...data },
    }).catch((error) => {
      dispatch(addTagFailure(error.message));
    });
    if (response) {
      dispatch(addTagSuccess(data));
    }
  };
};

export const updateTag = (data) => {
  return async (dispatch, getState) => {
    dispatch(loadingSpaces());

    const response = await axios({
      url: API_ADD_TAG + `/${data.id}`,
      method: 'put',
      data: { ...data },
    }).catch((error) => {
      dispatch(updateTagFailure(error.message));
    });
    if (response) {
      dispatch(updateTagSuccess(data));
    }
  };
};

export const deleteTag = (id) => {
  return async (dispatch, getState) => {
    dispatch(loadingSpaces());

    const response = await axios({
      url: API_ADD_TAG + `/${id}`,
      method: 'delete',
    }).catch((error) => {
      dispatch(deleteTagFailure(error.message));
    });
    if (response) {
      dispatch(deleteTagSuccess(id));
    }
  };
};

const loadingSpaces = () => ({
  type: LOADING_TAGS,
});

const getTagsSuccess = (data, query) => ({
  type: GET_TAGS_SUCCESS,
  payload: { data, query },
});

const getTagsFailure = (error) => ({
  type: GET_TAGS_FAILURE,
  payload: {
    error,
  },
});

const addTagSuccess = (data) => ({
  type: ADD_TAG_SUCCESS,
  payload: {
    ...data,
  },
});

const addTagFailure = (error) => ({
  type: ADD_TAG_FAILURE,
  payload: {
    error,
  },
});

const updateTagSuccess = (data) => ({
  type: UPDATE_TAG_SUCCESS,
  payload: {
    ...data,
  },
});

const updateTagFailure = (error) => ({
  type: UPDATE_TAG_FAILURE,
  payload: {
    error,
  },
});

const deleteTagSuccess = (id) => ({
  type: DELETE_TAG_SUCCESS,
  payload: id,
});

const deleteTagFailure = (error) => ({
  type: DELETE_TAG_FAILURE,
  payload: {
    error,
  },
});
