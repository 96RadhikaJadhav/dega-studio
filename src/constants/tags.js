//Actions
export const GET_TAGS_SUCCESS = 'GET_TAGS_SUCCESS';
export const GET_TAGS_FAILURE = 'GET_TAGS_FAILURE';
export const ADD_TAG_SUCCESS = 'ADD_TAG_SUCCESS';
export const ADD_TAG_FAILURE = 'ADD_TAG_FAILURE';
export const UPDATE_TAG_SUCCESS = 'UPDATE_TAG_SUCCESS';
export const UPDATE_TAG_FAILURE = 'UPDATE_TAG_FAILURE';
export const DELETE_TAG_SUCCESS = 'DELETE_TAG_SUCCESS';
export const DELETE_TAG_FAILURE = 'DELETE_TAG_FAILURE';
export const LOADING_TAGS = 'LOADING_TAGS';

//API
export const API_GET_TAGS = '/core/tags';
export const API_ADD_TAG = '/core/tags';

//Notification
