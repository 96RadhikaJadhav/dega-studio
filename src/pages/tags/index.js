import React from 'react';
import TagsList from './components/TagsList';
import { Space, Button } from 'antd';
import { Link } from 'react-router-dom';

function Tags() {
  return (
    <Space direction="vertical">
      <Link key="1" to="/tags/create">
        <Button>Create New</Button>
      </Link>
      <TagsList />
    </Space>
  );
}

export default Tags;
