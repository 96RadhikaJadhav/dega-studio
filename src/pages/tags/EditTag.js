import React from 'react';
import TagsCreateForm from './components/TagsCreateForm';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';
import { updateTag, getTag } from '../../actions/tags';
import { useHistory } from 'react-router-dom';
import { useParams } from 'react-router-dom';

function EditTag() {
  const history = useHistory();
  const { id } = useParams();

  const dispatch = useDispatch();
  const { tag, loading } = useSelector((state) => {
    return {
      tag: state.tags.details[id] ? state.tags.details[id] : null,
      loading: state.tags.loading,
    };
  });

  React.useEffect(() => {
    dispatch(getTag(id));
  }, [dispatch, id]);

  if (loading) return <Skeleton />;

  const onUpdate = (values) => {
    dispatch(updateTag({ ...tag, ...values }));
    history.push('/tags');
  };
  return <TagsCreateForm data={tag} onCreate={onUpdate} />;
}

export default EditTag;
