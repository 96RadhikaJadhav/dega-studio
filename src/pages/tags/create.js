import React from 'react';
import { useDispatch } from 'react-redux';
import { addTag } from '../../actions/tags';
import { useHistory } from 'react-router-dom';
import TagCreateForm from './components/TagsCreateForm';

function CreateTag() {
  const history = useHistory();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addTag(values));
    history.push('/tags');
    console.log('Received values of form: ', values);
  };
  return <TagCreateForm onCreate={onCreate} />;
}

export default CreateTag;
