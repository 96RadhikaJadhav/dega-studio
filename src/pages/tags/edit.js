import React from 'react';
import TagsCreateForm from './components/TagsCreateForm';
import { useDispatch, useSelector } from 'react-redux';
import { addTag } from '../../actions/tags';
import { useHistory } from 'react-router-dom';
import useQuery from '../../utils/useQuery';
import _ from 'lodash';

function EditTag() {
  const history = useHistory();
  const query = useQuery();
  const id = query.get('id');

  const dispatch = useDispatch();
  const { tag } = useSelector((state) => {
    const { tags, ...other } = state.tags;
    const tag = _.find(tags, { id: parseInt(id) });
    return {
      tag,
      ...other,
    };
  });

  const onCreate = (values) => {
    dispatch(addTag(values));
    history.push('/tags');
    console.log('Received values of form: ', values);
  };
  return <TagsCreateForm data={tag} onCreate={onCreate} />;
}

export default EditTag;
