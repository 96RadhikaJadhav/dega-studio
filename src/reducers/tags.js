import { GET_TAGS_SUCCESS, ADD_TAG_SUCCESS } from '../constants/tags';
import { LOADING_PAGE } from '../constants';

const initialState = {
  tags: [],
  loading: false,
};

export default function tagsReducer(state = initialState, action = {}) {
  if (!action.payload) {
    return state;
  }
  console.log('tags Reducer', state, action);
  switch (action.type) {
    case LOADING_PAGE:
      return {
        ...state,
        loading: true,
      };
    case GET_TAGS_SUCCESS:
      return {
        ...state,
        tags: action.payload.nodes,
        loading: false,
      };
    case ADD_TAG_SUCCESS:
      return {
        ...state,
        tags: [...state.tags, action.payload],
        loading: false,
      };
    default:
      return state;
  }
}
