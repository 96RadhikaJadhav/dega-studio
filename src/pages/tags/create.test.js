import React from 'react';
import { useHistory } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';

import '../../matchMedia.mock';
import CreateTag from './create';
import * as actions from '../../actions/tags';
import TagCreateForm from './components/TagsCreateForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));

jest.mock('../../actions/tags', () => ({
  getTags: jest.fn(),
  addTag: jest.fn(),
}));

describe('Tags create component', () => {
  let store;
  let mockedDispatch;

  store = mockStore({
    tags: {
      req: [],
      details: {},
      loading: true,
    },
  });
  store.dispatch = jest.fn(() => ({}));
  mockedDispatch = jest.fn(() => Promise.resolve({}));
  useDispatch.mockReturnValue(mockedDispatch);
  describe('snapshot testing', () => {
    it('should render the component', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <CreateTag />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    afterEach(() => {
      wrapper.unmount();
    });
    it('should call addTag', (done) => {
      actions.addTag.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CreateTag />
          </Provider>,
        );
      });
      wrapper.find(TagCreateForm).props().onCreate({ test: 'test' });
      setTimeout(() => {
        expect(actions.addTag).toHaveBeenCalledWith({ test: 'test' });
        expect(push).toHaveBeenCalledWith('/tags');
        done();
      }, 0);
    });
  });
});
