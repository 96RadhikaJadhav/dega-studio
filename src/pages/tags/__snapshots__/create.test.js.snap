// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags create component component testing should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="create-tag"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="create-tag_name"
        title="Tag Name"
      >
        Tag Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="create-tag_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="create-tag_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="create-tag_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="create-tag_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <textarea
            className="ant-input"
            id="create-tag_description"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Tags create component snapshot testing should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="create-tag"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="create-tag_name"
        title="Tag Name"
      >
        Tag Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="create-tag_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="create-tag_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="create-tag_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="create-tag_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <textarea
            className="ant-input"
            id="create-tag_description"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
